crates/host/src/wasmbus/handler.rs

```diff
@@ -169,13 +166,6 @@ impl wrpc_transport::Invoke for Handler {
     where
         P: AsRef<[Option<usize>]> + Send + Sync,
     {
-        // Reading a trace context should _never_ block because writing happens once at the beginning of a component
-        // invocation. If it does block here, it's a bug in the runtime, and it's better to deal with a
-        // disconnected trace than to block on the invocation for an extended period of time.
-        if let Ok(trace_context) = self.trace_ctx.try_read() {
```

Comment:
Could you help me understand why we had to replace this by setting the trace parent in each handler?

Reply:
Not Brooks but I helped him debug this. Essentially the trace context was being used at the top level wrpc server part so it was being shared among _all_ components (not reset for each handler like we do with link names). This was because the initial instantiate_component or serve (can't remember exactly which one) clones the trace_ctx from `Handler` once and then uses that on every invocation. This lead to interleaving of the traces since they were all sharing the same trace ctx

Reply:
Yep that's right, the trace context needs to be accessible per `Instance` rather than per Component/WrpcServer/Handler. So putting it in the store was the most obvious way I could see to do this, but happy to do it in an alternative way